homeassistant/components/recorder/statistics.py

```diff
@@ -2156,3 +2197,232 @@ def async_change_statistics_unit(
+    tables: tuple[type[Statistics | StatisticsShortTerm], ...] = (
+        Statistics,
+        StatisticsShortTerm,
+    )
+    try:
+        with session_scope(session=session_maker()) as session:
+            for table in tables:
+                _import_statistics_with_session(session, metadata, (statistics,), table)
+                stored_statistics = _statistics_during_period_with_session(
+                    hass,
+                    session,
+                    start_time,
+                    None,
+                    [statistic_id],
+                    "hour" if table == Statistics else "5minute",
+                    None,
+                    {"last_reset", "max", "mean", "min", "state", "sum"},
+                )
+                if not (stored_statistic := stored_statistics.get(statistic_id)):
+                    _LOGGER.warning(
+                        "Schema validation failed for table: %s", table.__tablename__
+                    )
+                    continue
+
+                check_columns(
+                    schema_errors,
+                    stored_statistic[0],
+                    statistics,
+                    ("max", "mean", "min", "state", "sum"),
+                    table.__tablename__,
+                    "double precision",
+                )
+                assert statistics["last_reset"]
+                check_columns(
+                    schema_errors,
+                    stored_statistic[0],
+                    {
+                        "last_reset": statistics["last_reset"],
+                        "start": statistics["start"],
+                    },
+                    ("start", "last_reset"),
+                    table.__tablename__,
+                    "µs precision",
+                )
+            _clear_statistics_with_session(session, statistic_ids=[statistic_id])
+    except Exception as exc:  # pylint: disable=broad-except
+        _LOGGER.exception("Error when validating DB schema: %s", exc)
+
+    return schema_errors
+
+
+def validate_db_schema(
+    hass: HomeAssistant, instance: Recorder, session_maker: Callable[[], Session]
+) -> set[str]:
+    """Do some basic checks for common schema errors caused by manual migration."""
+    schema_errors: set[str] = set()
+    schema_errors |= _validate_db_schema_utf8(instance, session_maker)
+    schema_errors |= _validate_db_schema(hass, instance, session_maker)
+    if schema_errors:
+        _LOGGER.debug(
+            "Detected statistics schema errors: %s", ", ".join(sorted(schema_errors))
+        )
+    return schema_errors
+
+
+def correct_db_schema(
+    engine: Engine, session_maker: Callable[[], Session], schema_errors: set[str]
+) -> None:
+    """Correct issues detected by validate_db_schema."""
+    from .migration import _modify_columns  # pylint: disable=import-outside-toplevel
+
+    if "statistics_meta.4-byte UTF-8" in schema_errors:
+        # Attempt to convert the table to utf8mb4
+        _LOGGER.warning(
```

Comment:
Should this be a notification ?

Reply:
oh I guess the frontend is not loaded yet.

Reply:
We could however create a repair issue